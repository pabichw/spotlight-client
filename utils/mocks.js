export const MOCKS = {
  PHOTOS: [
    {
      id: 0,
      name: "Test photo 1",
      url: "https://via.placeholder.com/500x500",
      created_at: "2022-07-29T23:54:56+00:00",
    },
    {
      id: 1,
      name: "Test photo 2",
      url: "https://via.placeholder.com/500x500",
      created_at: "2022-07-29T23:54:56+00:00",
    },
    {
      id: 2,
      name: "Test photo 3",
      url: "https://via.placeholder.com/500x500",
      created_at: "2022-07-29T23:54:56+00:00",
    },
  ],
};
