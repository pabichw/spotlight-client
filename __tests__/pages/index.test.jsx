import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import Home from '../../pages/index.js'
import { MOCKS } from '../../utils/mocks'

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
    };
  },
}));

describe('Main page', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules() // Most important - it clears the cache
    process.env = { ...OLD_ENV }; // Make a copy
  });


  afterEach(cleanup);

  it('renders header', () => {
    render(
      <Home />
    )

    const heading = screen.getByRole('heading', {
      name: 'Spotlight Wallpapers',
    })

    expect(heading).toBeInTheDocument()
  })

  it('renders list of given photos', async () => {
    const { findAllByTestId } = render(
      <Home photos={MOCKS.PHOTOS.slice(0, 3)}/>
    )

   const list = await findAllByTestId('photo-card'); 
   expect(list).toHaveLength(3);
  })
})
