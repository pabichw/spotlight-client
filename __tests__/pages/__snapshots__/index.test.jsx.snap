// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page render correct pages in pagination 1`] = `
<nav
  aria-label="pagination navigation"
  class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
  data-testid="pagination"
>
  <ul
    class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
  >
    <li>
      <button
        aria-label="Go to previous page"
        class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
          data-testid="NavigateBeforeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </button>
    </li>
    <li>
      <button
        aria-current="true"
        aria-label="page 1"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        1
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 2"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        2
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 3"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        3
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 4"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        4
      </button>
    </li>
    <li>
      <button
        aria-label="Go to page 5"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        5
      </button>
    </li>
    <li>
      <div
        class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
      >
        â€¦
      </div>
    </li>
    <li>
      <button
        aria-label="Go to page 8"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        8
      </button>
    </li>
    <li>
      <button
        aria-label="Go to next page"
        class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </button>
    </li>
  </ul>
</nav>
`;
